app/core/users/cruds_users.py

```diff
@@ -274,3 +275,122 @@ async def update_user_password_by_id(
         .values(password_hash=new_password_hash),
     )
     await db.commit()
+
+
+async def fusion_users(
+    db: AsyncSession,
+    user_kept_id: str,
+    user_deleted_id: str,
+):
+    """Fusion two users together
+    We use the user_kept_id as the user that will keep all the data
+    and the user_deleted_id as the user that will be deleted
+
+    The function will update all the foreign keys that reference the user_deleted_id
+    to reference the user_kept_id instead
+
+    If the user_deleted_id is referenced in a table where the user_kept_id is also referenced,
+    and the change would create a duplicate that violates a unique constraint, the row will be deleted
+
+    There are thre cases to consider:
+    1. The foreign key is not a primary key of the table:
+        In this case, we can update the row
+    2. The foreign key is one of the primary keys of the table:
+        In this case, we can't update the row without verifying that the new row doesn't already exist
+        So we collect all the primary keys of the table where the user_deleted_id is referenced and
+        all the primary keys of the table where the user_kept_id is referenced
+        We then update the row and check if the new row already exists:
+            a. If it doesn't, we update the row
+            b. If it does, we delete the row
+    """
+    foreign_keys: set[ForeignKey] = get_referencing_foreign_keys(
+        models_core.CoreUser.__table__,
+    )
+    user_id_fks: list[ForeignKey] = [
+        fk for fk in foreign_keys if fk.column == models_core.CoreUser.__table__.c.id
+    ]
```

Comment:
Could you add a comment explaining the purpose of this comprehension? 
I believe the idea is to keep all relationships related of a CoreUser `id`, and remove other relationships that are related to another column of CoreUser